synapse/federation/federation_server.py

```diff
@@ -811,12 +811,13 @@ def on_edu(self, edu_type, origin, content):
         if not handler:
             logger.warn("No handler registered for EDU type %s", edu_type)
 
-        try:
-            yield handler(origin, content)
-        except SynapseError as e:
-            logger.info("Failed to handle edu %r: %r", edu_type, e)
-        except Exception:
-            logger.exception("Failed to handle edu %r", edu_type)
+        with start_active_span_from_edu(content, "handle_edu"):
```

Comment:
AFAICT, this never worked because this PR serializes everything into `content["org.matrix.opentracing_context"]` but `start_active_span_from_edu` reads from `content["opentracing"]` (even at this point in time in the code).

https://github.com/matrix-org/synapse/blob/bbc23cdd04a2dfafefcbcb3264c2e89b50eacc04/synapse/logging/opentracing.py#L402